src/ripple/protocol/impl/RippleAddress.cpp

```diff
@@ -486,8 +488,23 @@ void RippleAddress::setAccountPublic (RippleAddress const& generator, int seq)
 }
 
 bool RippleAddress::accountPublicVerify (
-    uint256 const& uHash, Blob const& vucSig, ECDSA fullyCanonical) const
+    Blob const& message, Blob const& vucSig, ECDSA fullyCanonical) const
 {
+    if (vchData.size() == 33  &&  vchData[0] == 0xED)
+    {
+        if (vucSig.size() != 64)
+        {
+            return false;
+        }
+
+        uint8_t const* publicKey = &vchData[1];
+        uint8_t const* signature = &vucSig[0];
+
+        return ed25519_sign_open (message.data(), message.size(), publicKey, signature) == 0;
```

Comment:
Then I suggest a comment here: `// "sign_open" is an a synonym for "verify".`

Alternatively, could this be reworked to call `verify_ed25519` in AnyPublicKey.cpp?